cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -90,19 +88,23 @@ data ForkPoint
 startTimeOffsetSeconds :: DTC.NominalDiffTime
 startTimeOffsetSeconds = if OS.isWin32 then 90 else 15
 
--- | Like 'cardanoTestnet', but using defaults for all configuration files.
+-- | Like 'cardanoTestnet', but using 'ShelleyTestnetOptions' to obtain
+-- the genesis files, instead of passing them directly.
 -- See 'cardanoTestnet' for additional documentation.
 cardanoTestnetDefault
   :: ()
   => HasCallStack
   => CardanoTestnetOptions
+  -> ShelleyTestnetOptions
   -> Conf
   -> H.Integration TestnetRuntime
-cardanoTestnetDefault opts conf = do
-  AnyShelleyBasedEra sbe <- pure $ cardanoNodeEra cardanoDefaultTestnetOptions
+cardanoTestnetDefault testnetOptions shelleyOptions conf = do
+  AnyShelleyBasedEra sbe <- pure cardanoNodeEra
```

Comment:
This explains the change in the protocol parameters golden file: before we were not honoring the era passed within the `CardanoTestnetOptions` parameters. We were wrongly using `cardanoDefaultTestnetOptions`.

Now we take the era from the input parameter.